src/request.js

```diff
@@ -174,6 +177,33 @@ function requestOpts_to_axiosOpts( requestOptions, logger=defaultLogger ) {
     }
     axiosOptions.httpsAgent = new Https.Agent(agentOptions);
   }
+
+  // AWS4 options
+  if( axiosOptions.aws && axiosOptions.aws.key && axiosOptions.aws.secret ) {
+    // The URL host and path does not come naturally; it is parsed using the node.js `URL` library
+    // The `AWS4` library requires an uppercase method to run properly, unlike request-util
+    const urlObj = url.parse(axiosOptions.url, true);
+    const options = {
+      host: urlObj.host,
+      path: urlObj.path,
+      method: axiosOptions.method.toUpperCase(),
+      headers: axiosOptions.headers,
+      body: axiosOptions.data
+    };
+
+    // Add AWS4 `Authorization` and `X-Amz-Date` headers into axiosOptions headers
+    const signRes = aws4.sign(options, {
+      accessKeyId: axiosOptions.aws.key,
+      secretAccessKey: axiosOptions.aws.secret
+    });
+
+    // The original request module converted uppercase AWS4 headers to lowercase. Add in lowercase headers to ensure functionality
+    axiosOptions.headers = merge( axiosOptions.headers, {"authorization": signRes.headers.Authorization} );
+    axiosOptions.headers = merge( axiosOptions.headers, {"x-amz-date": signRes.headers[`X-Amz-Date`]} );
+    if (signRes.headers[`X-Amz-Security-Token`]) {
+      axiosOptions.headers = merge( axiosOptions.headers, {"x-amz-security-token": signRes.headers[`X-Amz-Security-Token`]} );
+    }
```

Comment:
`merge` is a pretty expensive operation -- it generates an entirely new object each time and recursively adds cloned attributes from the original object to it.  It should be used only when _intentionally_ needing to create a clone of the original object with changes (e.g. the initial clone so that `requestOpts_to_axiosOpts` does not alter the passed in `requestOptions` object as a side-effect of generating the `axiosOptions` return value).  In this case, cloning should not be necessary, and code similar to the original `request` module code would be more appropriate.